node/router/src/lib.rs

```diff
@@ -393,13 +405,19 @@ pub trait PeerPoolHandling<N: Network>: P2P {
     }
 
     /// Returns the list of candidate peers.
-    fn candidate_peers(&self) -> Vec<SocketAddr> {
+    fn get_candidate_peers(&self) -> Vec<CandidatePeer> {
         let banned_ips = self.tcp().banned_peers().get_banned_ips();
         self.peer_pool()
             .read()
             .iter()
             .filter_map(|(addr, peer)| {
-                (matches!(peer, Peer::Candidate(_)) && !banned_ips.contains(&addr.ip())).then_some(*addr)
+                if let Peer::Candidate(peer) = peer
+                    && !banned_ips.contains(&addr.ip())
```

Comment:
actually, this is no longer needed after https://github.com/ProvableHQ/snarkOS/pull/3905; I'll remove it shortly